### PR TITLE
docs: fix type checking in storybook

### PR DESCRIPTION
**Pull request by talkor**
https://monday.monday.com/boards/3532714909/pulses/6272877557
For example:
Screenshot 2024-08-12 at 21 11 22

It will also fail storybook build and prevent releasing corrupted versions

### CODE REVIEW

File: packages/core/src/hooks/useSwitch/index.ts
Comment (talkor) [inline packages/core/src/hooks/useSwitch/index.ts:1]: @YossiSaadi I had to revert this change, there were serious type issues otherwise. Maybe we can find other solution but I'm also ok with how it is now

File: packages/core/src/components/Steps/StepsHeader.tsx [hunk 1]
```diff
@@ -15,9 +15,9 @@ export interface StepsHeaderProps extends VibeComponentProps {
   onChangeActiveStep: (e: React.MouseEvent, stepIndex: number) => void;
   stepsCount: number;
   areNavigationButtonsHidden: boolean;
-  backButtonProps: ButtonProps;
-  nextButtonProps: ButtonProps;
-  finishButtonProps: ButtonProps;
+  backButtonProps: Partial<ButtonProps>;
+  nextButtonProps: Partial<ButtonProps>;
+  finishButtonProps: Partial<ButtonProps>;
```
Comment (talkor) [inline packages/core/src/components/Steps/StepsHeader.tsx:20]: I changed it in a different PR since Button's children is now mandatory but I think making it a Partial is a better solution instead of providing `children: ''` as it already have defaults

File: packages/core/src/components/Skeleton/__stories__/Skeleton.stories.tsx [hunk 2]
```diff
@@ -12,41 +12,37 @@ export default {
...
 const skeletonTemplate = (args: SkeletonProps) => {
+  return <Skeleton {...args} />;
+};
+
+export const Overview = {
+  render: skeletonTemplate.bind({}),
+  name: "Overview"
+};
+
+export const ComplexExample = () => {
```
Comment (talkor) [inline packages/core/src/components/Skeleton/__stories__/Skeleton.stories.tsx:23]: Had to extract the overview example to a different story to make the overview story a real playable example, need to verify with designers about the new story

File: packages/core/src/hooks/useSwitch/index.ts
Comment (YossiSaadi) [reply packages/core/src/hooks/useSwitch/index.ts:1]: Not sure I understand what exactly was reverted here 🫣

File: packages/core/.storybook/main.ts [hunk 3]
```diff
@@ -39,6 +39,12 @@ const config: StorybookConfig = {
   docs: {
     autodocs: false
   },
+  typescript: {
```
Comment (YossiSaadi) [inline packages/core/.storybook/main.ts:42]: It read that simple? 🤦🏻‍♂️

File: packages/core/src/hooks/useSwitch/index.ts
Comment (talkor) [reply packages/core/src/hooks/useSwitch/index.ts:1]: [This](https://github.com/mondaycom/vibe/pull/2341)

File: packages/core/.storybook/main.ts (hunk 3 above)
Comment (talkor) [reply packages/core/.storybook/main.ts:42]: Looks like it haha

File: packages/core/src/components/Skeleton/__stories__/Skeleton.stories.tsx
Comment (shaharzil) [inline packages/core/src/components/Skeleton/__stories__/Skeleton.stories.tsx:1]: Maybe add the live edit if we're already touching this file?

**Review by shaharzil — APPROVED** [review]
